### PR TITLE
Commit hook

### PR DESCRIPTION
**Pull request by schalkms**
This is the first version. The commit hook works locally.
Should detekt also include a server side hook?
https://git-scm.com/book/uz/v2/Customizing-Git-An-Example-Git-Enforced-Policy

### CODE REVIEW

File: commit-msg [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+msg=$1
+
+# checks if the first character of the commit message is an uppercase letter
+uppercase='^[A-Z]'
+if ! head -1 "$msg" | grep -Eq "$uppercase"; then
+  echo "Aborting commit. The commit message does not start with an uppercase letter." >&2
+  exit 1
+fi
+
+# checks if the second line of the commit message is empty
```
Comment (vanniktech) [inline commit-msg]: What's that one used for?

File: commit-msg [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+len=`awk 'END{print NR}' $msg`
+if [ "$len" -ge 2 ]; then
+  line=`sed -n 2p $msg`
+  if [ "$line" != "" ]; then
+    echo "Aborting commit. The commit message does not have a empty second line." >&2
+    exit 1
+  fi
+fi
+
+# checks if all lines of the commit message do not exceed the length 70
```
Comment (vanniktech) [inline commit-msg]: Do we really want to enforce this?

**Comment by schalkms** [comment]
@vanniktech 
This is just a proposal. There is a discussion in #797

**Comment by Mauin** [comment]
Not really sure this is going to give us much benefit. As @vanniktech already mentioned just using "Squash and Merge" instead of simply merging PRs we will get one well-named commit with the title of the PR (which should have good, descriptive names and we can even rename them if necessary). This commit hook will add another tripwire for people trying to quickly contribute to detekt. 
I think we shouldn't make it harder than necessary. GitHub can take care of squashing, merging and renaming the appropriate commit for us, let's not put that burden onto people trying to contribute.

File: commit-msg (hunk 1 above)
Comment (arturbosch) [reply commit-msg]: git commit should be written as:

the title with issue id

longer description if necessary.

I think it checks if the second line is empty so git detects it as the separator of the detailed message

File: commit-msg [hunk 3]
```diff
@@ -0,0 +1,35 @@
+#!/bin/sh
+#
+# This hook script checks the commit log message.
+# Called by "git commit" with one argument, the name of the file that has the commit message.
+
+msg=$1
+
+# checks if the first character of the commit message is an uppercase letter
```
Comment (arturbosch) [inline commit-msg]: personally I always start with a capital letter so :+1: from me

File: commit-msg (hunk 2 above)
Comment (arturbosch) [reply commit-msg]: I also dont think we should enforce this. Ok for the other two

**Pull request merged**